content/blog/calisto-mathias/2024-05-15_[gsoc_2024]_enhancing_tracker_next-gen_find_window_and_functionalities.md

```diff
@@ -0,0 +1,46 @@
+
+I plan to integrate new features, such as targeted directory searches for files to improve efficiency, and displaying non-indexed attributes in the column list view as part of the results, enhancing the search experience. However, users won’t be able to add search terms specifically for these attributes.
+
+# Goals
+
+## Rework the User Interface for the Find Panel
+
+The foundation of the project centers on developing a contemporary and user-friendly find panel, leveraging classes from the Interface Kit to enhance the search experience. Depending on time feasibility, exploring the creation of a custom column class to display more complex views in the Column Types could be considered. The find panel would comprise a Column List View, featuring selected search attributes with corresponding input fields positioned above the relevant columns, streamlining the process of executing queries and locating files faster and more efficiently.
+
+## Introducing Directory-Specific Search Capability
+
+The project also aims to introduce the capability of executing search queries within specific directories, in addition to searching across entire volumes. This feature enhancement will offer users greater flexibility and precision in their file searches, further enhancing the overall functionality and usability of the find panel.
+
+## Inclusion of Non-Indexed Attributes in Search Results
+
+In further enriching the Find Panel’s functionality, the project includes the incorporation of non-indexed attributes in search results. Recognizing that these attributes often hold valuable information for users to identify their files, this enhancement aims to provide a more comprehensive search experience. Each non-indexed attribute will also be displayed in its specific column along with the result entries, complemented by a singular disabled search input field above it. This design element serves to indicate that these attributes cannot be used for direct searching, yet their presence offers users additional context and insight into their files.
+
+## Tracking Progress of This Project
+
+Right now, my mentor Neils and I are figuring out the best way to keep track of the changes I make to the code. Since the project involves a lot of reworking and adding new features, it’s important to have a plan for how these changes will fit into the Haiku software smoothly. As soon as we come to a good workflow, it will be uploaded on this blog-post. Stay tuned for updates!
```

Comment:
```suggestion
Right now, my mentor Niels and I are figuring out the best way to keep track of the changes I make to the code. Since the project involves a lot of reworking and adding new features, it’s important to have a plan for how these changes will fit into the Haiku software smoothly. As soon as we come to a good workflow, it will be uploaded on this blog-post. Stay tuned for updates!
```